### PR TITLE
Migrate pub in flutter_tools to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/build_system/source_test.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ import 'package:flutter_tools/src/build_system/exceptions.dart';
 import 'package:flutter_tools/src/globals.dart' as globals;
 
 import '../../src/common.dart';
-import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/build_system/source_test.dart:16]: Not related, but noticed this file was just using the exported `fake_process_manager`, which is null safe.

**Comment by jonahwilliams** [comment]
Trade you for https://github.com/flutter/flutter/pull/80462

**Comment by jmagman** [comment]
> Trade you for #80462

Done.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**